nixos/modules/services/misc/klipper.nix

```diff
@@ -69,6 +67,26 @@ in
           for supported values.
         '';
       };
+
+      firmware = mkOption {
+        description = "Firmwares klipper should manage";
+        default = { };
+        type = with types; attrsOf
+          (submodule {
+            options = {
+              enableFlashing = mkEnableOption ''
+                automatic flashing of firmware to microcontroller
+
+                WARNING: Be careful with this option. Enabling this will automatically flash your microcontroller on e.g. nixos-rebuild.
+                This can potentially brick your microcontroller. Alternatively, use `klipper-flash-$mcu` to flash manually.
+              '';
+              firmwareConfig = mkOption {
+                type = path;
+                description = "Path to firmware config which is generated using `klipper-genconf`";
+              };
```

Comment:
```suggestion
              configFile = mkOption {
                type = path;
                description = "Path to firmware config which is generated using `klipper-genconf`";
              };
```

Reply:
Done in cc3dcdbf0c3b3c263acf839e9171d7270e4f92ce, as it requires changes at other places as well.